File: src/main/java/com/maxciv/infer/plugin/process/DefineBuildTool.java
package com.maxciv.infer.plugin.process;

import com.intellij.openapi.project.Project;

/**
 * @author maxim.oleynik
 * @since 29.11.2018
 */
public final class DefineBuildTool {

    private DefineBuildTool() {
    }

    public static BuildTools defineFor(Project project) {
        if (project.getBaseDir().findChild("pom.xml") != null) {
            return BuildTools.MAVEN;
        }
        if (project.getBaseDir().findChild("build.gradle") != null) {
            return BuildTools.GRADLE;
        }
        return BuildTools.DEFAULT;
    }
}

File: src/main/kotlin/com/maxciv/infer/plugin/process/DefineBuildTool.kt
package com.maxciv.infer.plugin.process

import com.intellij.openapi.project.Project

/**
 * @author maxim.oleynik
 * @since 29.11.2018
 */
object DefineBuildTool {

    fun defineFor(project: Project): BuildTools {
        return when {
            project.baseDir.findChild("pom.xml") != null -> BuildTools.MAVEN
            project.baseDir.findChild("gradlew") != null -> BuildTools.GRADLEW
            project.baseDir.findChild("build.gradle") != null -> BuildTools.GRADLE
            else -> BuildTools.DEFAULT
        }
    }
}
